File: ruby/hamming/hamming.rb
# frozen_string_literal: true

# This function returns the Hamming distance
class Hamming
  def self.compute(strand_one, strand_two)
    raise ArgumentError if strand_one.length != strand_two.length

    (0..strand_one.size).count { |i| + 1 if strand_one[i] != strand_two[i] }
  end
end

File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

func Distance(strandOne, strandTwo string) (int, error) {
	if len(strandOne) != len(strandTwo) {
		return 0, errors.New("disallow strand")
	}

	var counter int
	for index := range strandOne {
		if strandOne[index] != strandTwo[index] {
			counter++
		}
	}
	return counter, nil
}
